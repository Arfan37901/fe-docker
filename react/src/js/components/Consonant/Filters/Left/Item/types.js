import {
    number,
    bool,
    string,
    func,
    arrayOf,
    shape,
} from 'prop-types';

import { FilterItemType } from '../../../types/config';

// eslint-disable-next-line import/prefer-default-export
export const ItemType = {
    icon: string,
    isOpened: bool,
    id: string.isRequired,
    clearFilterText: string,
    name: string.isRequired,
    onCheck: func.isRequired,
    onClick: func.isRequired,
    numItemsSelected: number,
    results: number.isRequired,
    onClearAll: func.isRequired,
    items: arrayOf(shape(FilterItemType)).isRequired,
};
