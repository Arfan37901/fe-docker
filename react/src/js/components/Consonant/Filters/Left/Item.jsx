import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { Items } from './Items';
import { GroupFooter } from './Mobile-Only/GroupFooter';
import { SelectedItem as DesktopSelectedItem } from './Desktop-Only/SelectedItem';

import { useConfig } from '../../Helpers/hooks';

/**
 * Left filter
 *
 * @component
 * @example
 * const props= {
    name: String,
    icon: String,
    id: String,
    items: Array,
    numItemsSelected: Number,
    isOpened: Boolean,
    onCheck: Function,
    onClick: Function,
    onClearAll: Function,
    results: Number,
    clearFilterText: String,
 * }
 * return (
 *   <Item {...props}/>
 * )
 */
const Item = (props) => {
    const {
        name,
        icon,
        id,
        items,
        numItemsSelected,
        isOpened,
        onCheck,
        onClick,
        onClearAll,
        results,
        clearFilterText,
    } = props;

    const getConfig = useConfig();

    /**
     **** Authored Configs ****
     */
    const mobileGroupTotalResultsText =
        getConfig('filterPanel', 'i18n.leftPanel.mobile.group.totalResultsText').replace('{total}', results);
    const applyBtnText = getConfig('filterPanel', 'i18n.leftPanel.mobile.group.applyBtnText');
    const doneBtnText = getConfig('filterPanel', 'i18n.leftPanel.mobile.group.doneBtnText');

    /**
     **** Constants ****
     */

    /**
     * Text of the left filter footer button on the mobile and tablet breakpoints:
     * whether the "Apply changes" text should be shown or "Done"
     * @type {String}
     */
    const buttonText = numItemsSelected > 0 ? applyBtnText : doneBtnText;

    /**
     * Text indicating the quantity of the selected left filter options
     * @type {String}
     */
    const dataQtyTxt = numItemsSelected > 0 ? `+${numItemsSelected}` : '';

    /**
     * Whether the left filter selected options badge should be shown on desktop breakpoint
     * @type {Boolean}
     */
    const shouldRenderSelectedBadge = numItemsSelected > 0;

    /**
     * Array of the left filter selected options
     * @type {Array}
     */
    const selectedFilters = items.filter(item => item.selected);

    /**
     * Array of the selected filters labels shown on mobile and tablet breakpoints
     * @type {Array}
     */
    const mobileTagsSelectedText = selectedFilters.map((item, index) => {
        const lastItem = index === selectedFilters.length - 1;
        return (!lastItem ? `${item.label}, ` : item.label);
    });

    /**
     * Handles toggling the opened/closed state of the left filter
     * when the filter is clicked
     * @param {ClickEvent} e
     * @listens ClickEvent
     */
    const handleClick = (e) => {
        e.preventDefault();
        onClick(id);
    };

    /**
     * Handles unselection of the left filter options
     * when the "Clear filter options" button is clicked
     * @param {ClickEvent} e
     * @listens ClickEvent
     */
    const handleClear = () => onClearAll(id);

    /**
     * Handles toggling the selected/unselected state of the left filter option
     * when the filter option is checked/unchecked
     * @param {ChangeEvent} e
     * @listens ChangeEvent
     */
    const handleCheck = (evt) => {
        evt.stopPropagation();
        onCheck(id, evt.target.value, evt.target.checked);
    };

    /**
     * Class name for the left filter:
     * whether the left filter is opened
     * @type {String}
     */
    const leftFilterClassName = classNames({
        'consonant-left-filter': true,
        'consonant-left-filter_opened': isOpened,
    });

    return (
        <div
            data-testid="filter-item"
            className={leftFilterClassName}>
            <div
                className="consonant-left-filter--inner">
                <h3
                    className="consonant-left-filter--name">
                    {icon &&
                        <img
                            src={icon}
                            width="16"
                            alt=""
                            loading="lazy" />
                    }
                    <button
                        type="button"
                        data-testid="filter-item-link"
                        className="consonant-left-filter--link"
                        onClick={handleClick}
                        tabIndex="0">
                        {name}
                        <div
                            className="consonant-left-filter--selected-items-qty"
                            data-qty={dataQtyTxt}>
                            {mobileTagsSelectedText}
                        </div>
                    </button>
                </h3>
                {
                    shouldRenderSelectedBadge &&
                        <DesktopSelectedItem
                            handleClear={handleClear}
                            numItemsSelected={numItemsSelected} />
                }
                {
                    <Items
                        items={items}
                        handleCheck={handleCheck} />
                }
                {
                    <GroupFooter
                        ctaText={buttonText}
                        handleClick={handleClick}
                        clearFilterText={clearFilterText}
                        handleClear={handleClear}
                        numItemsSelected={numItemsSelected}
                        mobileGroupTotalResultsText={mobileGroupTotalResultsText} />
                }
            </div>
        </div>
    );
};

Item.propTypes = {
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    icon: PropTypes.string,
    onCheck: PropTypes.func.isRequired,
    onClick: PropTypes.func.isRequired,
    onClearAll: PropTypes.func.isRequired,
    items: PropTypes.arrayOf(PropTypes.object).isRequired,
    numItemsSelected: PropTypes.number,
    isOpened: PropTypes.bool,
    results: PropTypes.number.isRequired,
    clearFilterText: PropTypes.string,
};

Item.defaultProps = {
    icon: '',
    isOpened: false,
    numItemsSelected: 0,
    clearFilterText: 'Clear',
};

export default Item;
