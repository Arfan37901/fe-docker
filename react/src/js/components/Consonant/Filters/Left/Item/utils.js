// eslint-disable-next-line import/prefer-default-export
export const selector = ({
    filterPanel: {
        i18n: {
            leftPanel: {
                mobile: {
                    group: {
                        doneBtnText,
                        applyBtnText,
                        totalResultsText,
                    },
                },
            },
        },
    },
}) => ({
    doneButtonText: doneBtnText,
    applyButtonText: applyBtnText,
    totalResultsTextTemplate: totalResultsText,
});
