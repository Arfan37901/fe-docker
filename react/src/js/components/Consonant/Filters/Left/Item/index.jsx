import React, {
    useMemo,
    useCallback,
} from 'react';
import classNames from 'classnames';

import { Items } from '../Items';
import { selector } from './utils';
import { ItemType } from './types';
import { If } from '../../../Common';
import { template } from '../../../Helpers/general';
import { GroupFooter } from '../Mobile-Only/GroupFooter';
import { useConfigSelector } from '../../../Helpers/hooks';
import { SelectedItem as DesktopSelectedItem } from '../Desktop-Only/SelectedItem';

const defaultProps = {
    icon: '',
    isOpened: false,
    numItemsSelected: 0,
    clearFilterText: '',
};

/**
 * Left filter
 *
 * @component
 * @example
 * const props= {
    name: String,
    icon: String,
    id: String,
    items: Array,
    numItemsSelected: Number,
    isOpened: Boolean,
    onCheck: Function,
    onClick: Function,
    onClearAll: Function,
    results: Number,
    clearFilterText: String,
 * }
 * return (
 *   <Item {...props}/>
 * )
 */
const Item = ({
    id,
    name,
    icon,
    items,
    onCheck,
    onClick,
    results,
    isOpened,
    onClearAll,
    clearFilterText,
    numItemsSelected,
}) => {
    /**
     **** Authored Configs ****
     */

    const {
        doneButtonText,
        applyButtonText,
        totalResultsTextTemplate,
    } = useConfigSelector(selector);

    const totalResultsText = template(totalResultsTextTemplate, { total: results });

    /**
     **** Constants ****
     */

    /**
     * Text of the left filter footer button on the mobile and tablet breakpoints:
     * whether the "Apply changes" text should be shown or "Done"
     * @type {String}
     */
    const buttonText = numItemsSelected > 0 ? doneButtonText : applyButtonText;

    /**
     * Text indicating the quantity of the selected left filter options
     * @type {String}
     */
    const dataQtyTxt = numItemsSelected > 0 ? `+${numItemsSelected}` : '';

    /**
     * Whether the left filter selected options badge should be shown on desktop breakpoint
     * @type {Boolean}
     */
    const shouldRenderSelectedBadge = numItemsSelected > 0;

    /**
     * Array of the left filter selected options
     * @type {Array}
     */
    const selectedFilters = useMemo(() => items.filter(({ selected }) => selected), [items]);

    /**
     * Array of the selected filters labels shown on mobile and tablet breakpoints
     * @type {Array}
     */
    const mobileTagsSelectedText = useMemo(() => selectedFilters.map(({ label }, index) => {
        const lastItem = index === selectedFilters.length - 1;

        return `${label}${lastItem ? '' : ', '}`;
    }), [selectedFilters]);

    /**
     * Handles toggling the opened/closed state of the left filter
     * when the filter is clicked
     * @param {ClickEvent} event
     * @listens ClickEvent
     */
    const handleClick = useCallback((event) => {
        event.preventDefault();

        onClick(id);
    }, [id, onClick]);

    /**
     * Handles unselection of the left filter options
     * when the "Clear filter options" button is clicked
     */
    const handleClear = useCallback(() => {
        onClearAll(id);
    }, [id, onClearAll]);

    /**
     * Handles toggling the selected/unselected state of the left filter option
     * when the filter option is checked/unchecked
     * @param {ChangeEvent} event
     * @listens ChangeEvent
     */
    const handleCheck = useCallback((event) => {
        const { target: { value, checked } } = event;

        event.stopPropagation();

        onCheck(id, value, checked);
    }, [id, onCheck]);

    /**
     * Class name for the left filter:
     * whether the left filter is opened
     * @type {String}
     */
    const leftFilterClassName = classNames({
        'consonant-left-filter': true,
        'consonant-left-filter_opened': isOpened,
    });

    return (
        <div
            data-testid="filter-item"
            className={leftFilterClassName}>
            <div
                className="consonant-left-filter--inner">
                <h3
                    className="consonant-left-filter--name">
                    <If condition={Boolean(icon)}>
                        <img
                            alt=""
                            src={icon}
                            width="16"
                            loading="lazy" />
                    </If>
                    <button
                        tabIndex="0"
                        type="button"
                        onClick={handleClick}
                        data-testid="filter-item-link"
                        className="consonant-left-filter--link">
                        {name}
                        <div
                            data-qty={dataQtyTxt}
                            className="consonant-left-filter--selected-items-qty">
                            {mobileTagsSelectedText}
                        </div>
                    </button>
                </h3>
                <If condition={Boolean(shouldRenderSelectedBadge)}>
                    <DesktopSelectedItem
                        handleClear={handleClear}
                        numItemsSelected={numItemsSelected} />
                </If>
                <Items
                    items={items}
                    handleCheck={handleCheck} />
                <GroupFooter
                    ctaText={buttonText}
                    handleClick={handleClick}
                    handleClear={handleClear}
                    clearFilterText={clearFilterText}
                    numItemsSelected={numItemsSelected}
                    mobileGroupTotalResultsText={totalResultsText} />
            </div>
        </div>
    );
};

Item.propTypes = ItemType;
Item.defaultProps = defaultProps;

export default Item;
