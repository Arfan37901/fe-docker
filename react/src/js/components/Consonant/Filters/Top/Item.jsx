import classNames from 'classnames';
import PropTypes from 'prop-types';
import React, { useCallback } from 'react';
import { useExpandable } from '../../../../utils/hooks';

const clipWrapperItemsCount = 9;
const TopFilterItem = ({
    name,
    id,
    items,
    numItemsSelected,
    onCheck,
    onClearAll,
    results,
    clearFilterText,
}) => {
    const [openDropdown, handleToggle] = useExpandable(id);
    const isOpened = openDropdown === id;

    const stopPropagation = useCallback(e => e.stopPropagation(), []);

    const handleClear = useCallback((e) => {
        e.stopPropagation();
        onClearAll(id);
    }, [onClearAll, id]);

    const handleCheck = useCallback((e) => {
        e.stopPropagation();
        onCheck(id, e.target.value, e.target.checked);
    }, [onCheck, id]);

    const itemsComponent = (
        <ul
            data-testid="filter-group"
            className={items.length >= clipWrapperItemsCount ?
                'consonant-top-filter--items consonant-top-filter--items_clipped' :
                'consonant-top-filter--items'
            }>
            {items.map(item => (
                <li
                    key={item.id}
                    data-testid="filter-group-item"
                    className="consonant-top-filter--items-item">
                    {/* eslint-disable-next-line max-len */}
                    {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-noninteractive-element-interactions */}
                    <label htmlFor={item.id} className="consonant-top-filter--items-item-label" onClick={stopPropagation}>
                        <input
                            data-testid="list-item-checkbox"
                            id={item.id}
                            value={item.id}
                            type="checkbox"
                            onChange={handleCheck}
                            checked={item.selected}
                            tabIndex="0" />
                        <span className="consonant-top-filter--items-item-checkmark" />
                        <span className="consonant-top-filter--items-item-name">{item.label}</span>
                    </label>
                </li>
            ))}
        </ul>
    );
    const footerComponent = (
        <div className="consonant-top-filter--footer">
            <span className="consonant-top-filter--footer-res-qty">{results} results</span>
            {numItemsSelected > 0 &&
            <button
                data-testid="clear-btn"
                type="button"
                onClick={handleClear}
                className="consonant-top-filter--footer-clear-btn"
                tabIndex="0">{clearFilterText}
            </button>}
            <button
                type="button"
                onClick={handleToggle}
                className="consonant-top-filter--footer-btn"
                tabIndex="0">
                {numItemsSelected > 0 ? 'Apply' : 'Done'}
            </button>
        </div>
    );

    const containerClassnames = classNames('consonant-top-filter', {
        'consonant-top-filter_opened': isOpened,
        'consonant-top-filter_selected': items.filter(i => i.selected).length > 0 && !isOpened,
    });

    return (
        <div data-testid="filter-item" className={containerClassnames}>
            <div className="consonant-top-filter--inner">
                <h3 className="consonant-top-filter--name">
                    <button
                        type="button"
                        className="consonant-top-filter--link"
                        data-testid="filter-group-btn"
                        onClick={handleToggle}
                        tabIndex="0">
                        {name}
                        <span className="consonant-top-filter--selcted-items-qty">
                            {items.filter(item => item.selected).length || null}
                        </span>
                    </button>
                </h3>
                <div className="consonant-top-filter--selcted-items">
                    <div className="consonant-top-filter--absolute-wrapper">
                        {itemsComponent}
                        {items.length >= clipWrapperItemsCount && <aside className="consonant-top-filter--bg" />}
                        {footerComponent}
                    </div>
                </div>
            </div>
        </div>
    );
};

export default TopFilterItem;

TopFilterItem.propTypes = {
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    onCheck: PropTypes.func.isRequired,
    onClearAll: PropTypes.func.isRequired,
    items: PropTypes.arrayOf(PropTypes.object).isRequired,
    numItemsSelected: PropTypes.number,
    results: PropTypes.number.isRequired,
    clearFilterText: PropTypes.string,
};

TopFilterItem.defaultProps = {
    numItemsSelected: 0,
    clearFilterText: 'Clear',
};
