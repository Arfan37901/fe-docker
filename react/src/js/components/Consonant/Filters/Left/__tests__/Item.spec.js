import { screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Item from '../Item';

import {
    DEFAULT_PROPS,
    selectedAllItems,
} from '../../../Helpers/Testing/Constants/FilterItem';

import makeSetup from '../../../Helpers/Testing/Utils/Settings';

const setup = makeSetup(Item, DEFAULT_PROPS);

describe('Consonant/FilterItem', () => {
    test('should render without item count badge', () => {
        setup();

        const badgeElement = screen.queryByTestId('item-badge');

        expect(badgeElement).toBeNull();
    });
    test('should render with item count badge', () => {
        setup(selectedAllItems);

        const badgeElement = screen.queryByTestId('item-badge');

        expect(badgeElement).not.toBeNull();

        expect(badgeElement).toHaveTextContent(String(selectedAllItems.itemsSelected));
    });
    test('should render all list items', () => {
        const { props: { items } } = setup();

        const filterItemElement = screen.queryAllByTestId('filter-item');

        expect(filterItemElement).toHaveLength(items.length);
    });

    describe('Check snapshots', () => {
        test('should renders correctly when closed', () => {
            const { tree } = setup();

            expect(tree).toMatchSnapshot();
        });
        test('should renders correctly when opened', () => {
            const { tree } = setup({ isOpened: true });

            expect(tree).toMatchSnapshot();
        });
    });

    describe('Interaction with UI', () => {
        test('should call onCheck', () => {
            const { props: { onCheck } } = setup();

            const [checkboxElement] = screen.queryAllByTestId('list-item-checkbox');

            expect(checkboxElement).toBeDefined();

            fireEvent.click(checkboxElement);

            expect(onCheck).toBeCalled();
        });
        test('should call onClick', () => {
            const { props: { onClick, name } } = setup();

            const itemLinkElement = screen.getByText(name);

            expect(itemLinkElement).not.toBeNull();

            fireEvent.click(itemLinkElement);

            expect(onClick).toBeCalled();
        });
        test('should call onClearAll', () => {
            const { props: { onClearAll } } = setup(selectedAllItems);

            const badgeElement = screen.queryByTestId('item-badge');

            expect(badgeElement).not.toBeNull();

            fireEvent.click(badgeElement);

            expect(onClearAll).toBeCalled();
        });
    });
});
