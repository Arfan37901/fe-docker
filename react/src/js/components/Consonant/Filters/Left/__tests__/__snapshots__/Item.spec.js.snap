// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/FilterItem Check snapshots should renders correctly when closed 1`] = `
<div
  className="consonant-left-filter"
  data-testid="filter-item"
>
  <div
    className="consonant-left-filter--inner"
  >
    <h3
      className="consonant-left-filter--name"
    >
      <img
        alt=""
        loading="lazy"
        src="https://upload.wikimedia.org/wikipedia/commons/e/e0/Check_green_icon.svg"
        width="16"
      />
      <button
        className="consonant-left-filter--link"
        data-testid="filter-item-link"
        onClick={[Function]}
        tabIndex="0"
        type="button"
      >
        Topic
        <div
          className="consonant-left-filter--selected-items-qty"
          data-qty=""
        >
          
          
          
          
          
          
          
          
          
          
        </div>
      </button>
    </h3>
    <ul
      className="consonant-left-filter--items"
      data-testid="filter-group"
    >
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="we-retail:activity/hiking"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="we-retail:activity/hiking"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="we-retail:activity/hiking"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Business Continuity
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="we-retail:activity/biking"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="we-retail:activity/biking"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="we-retail:activity/biking"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Creativity and Design
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="adobe-com-enterprise:topic/customer-intelligence"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="adobe-com-enterprise:topic/customer-intelligence"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="adobe-com-enterprise:topic/customer-intelligence"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Customer Intelligence
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="adobe-com-enterprise:topic/data-management-platform"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="adobe-com-enterprise:topic/data-management-platform"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="adobe-com-enterprise:topic/data-management-platform"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Data Management Platform
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="adobe-com-enterprise:topic/digital-foundation"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="adobe-com-enterprise:topic/digital-foundation"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="adobe-com-enterprise:topic/digital-foundation"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Digital Foundation
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="adobe-com-enterprise:topic/digital-trends"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="adobe-com-enterprise:topic/digital-trends"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="adobe-com-enterprise:topic/digital-trends"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Digital Trends
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="adobe-com-enterprise:topic/document-management"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="adobe-com-enterprise:topic/document-management"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="adobe-com-enterprise:topic/document-management"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Document Management
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="adobe-com-enterprise:topic/marketing-automation"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="adobe-com-enterprise:topic/marketing-automation"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="adobe-com-enterprise:topic/marketing-automation"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Marketing Automation
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="adobe-com-enterprise:topic/personalization"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="adobe-com-enterprise:topic/personalization"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="adobe-com-enterprise:topic/personalization"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Personalization
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="adobe-com-enterprise:topic/Stock"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="adobe-com-enterprise:topic/Stock"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="adobe-com-enterprise:topic/Stock"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Stock
          </span>
        </label>
      </li>
    </ul>
    <div
      className="consonant-left-filter--footer"
    >
      <span
        className="consonant-left-filter--footer-res-qty"
      >
        0
         results
      </span>
      <button
        className="consonant-left-filter--footer-btn"
        onClick={[Function]}
        type="button"
      >
        Done
      </button>
    </div>
  </div>
</div>
`;

exports[`Consonant/FilterItem Check snapshots should renders correctly when opened 1`] = `
<div
  className="consonant-left-filter consonant-left-filter_opened"
  data-testid="filter-item"
>
  <div
    className="consonant-left-filter--inner"
  >
    <h3
      className="consonant-left-filter--name"
    >
      <img
        alt=""
        loading="lazy"
        src="https://upload.wikimedia.org/wikipedia/commons/e/e0/Check_green_icon.svg"
        width="16"
      />
      <button
        className="consonant-left-filter--link"
        data-testid="filter-item-link"
        onClick={[Function]}
        tabIndex="0"
        type="button"
      >
        Topic
        <div
          className="consonant-left-filter--selected-items-qty"
          data-qty=""
        >
          
          
          
          
          
          
          
          
          
          
        </div>
      </button>
    </h3>
    <ul
      className="consonant-left-filter--items"
      data-testid="filter-group"
    >
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="we-retail:activity/hiking"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="we-retail:activity/hiking"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="we-retail:activity/hiking"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Business Continuity
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="we-retail:activity/biking"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="we-retail:activity/biking"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="we-retail:activity/biking"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Creativity and Design
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="adobe-com-enterprise:topic/customer-intelligence"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="adobe-com-enterprise:topic/customer-intelligence"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="adobe-com-enterprise:topic/customer-intelligence"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Customer Intelligence
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="adobe-com-enterprise:topic/data-management-platform"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="adobe-com-enterprise:topic/data-management-platform"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="adobe-com-enterprise:topic/data-management-platform"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Data Management Platform
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="adobe-com-enterprise:topic/digital-foundation"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="adobe-com-enterprise:topic/digital-foundation"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="adobe-com-enterprise:topic/digital-foundation"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Digital Foundation
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="adobe-com-enterprise:topic/digital-trends"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="adobe-com-enterprise:topic/digital-trends"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="adobe-com-enterprise:topic/digital-trends"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Digital Trends
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="adobe-com-enterprise:topic/document-management"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="adobe-com-enterprise:topic/document-management"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="adobe-com-enterprise:topic/document-management"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Document Management
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="adobe-com-enterprise:topic/marketing-automation"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="adobe-com-enterprise:topic/marketing-automation"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="adobe-com-enterprise:topic/marketing-automation"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Marketing Automation
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="adobe-com-enterprise:topic/personalization"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="adobe-com-enterprise:topic/personalization"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="adobe-com-enterprise:topic/personalization"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Personalization
          </span>
        </label>
      </li>
      <li
        className="consonant-left-filter--items-item"
        data-testid="filter-group-item"
      >
        <label
          className="consonant-left-filter--items-item-label"
          htmlFor="adobe-com-enterprise:topic/Stock"
        >
          <input
            checked={undefined}
            data-testid="list-item-checkbox"
            id="adobe-com-enterprise:topic/Stock"
            onChange={[Function]}
            tabIndex="0"
            type="checkbox"
            value="adobe-com-enterprise:topic/Stock"
          />
          <span
            className="consonant-left-filter--items-item-checkmark"
          />
          <span
            className="consonant-left-filter--items-item-name"
          >
            Stock
          </span>
        </label>
      </li>
    </ul>
    <div
      className="consonant-left-filter--footer"
    >
      <span
        className="consonant-left-filter--footer-res-qty"
      >
        0
         results
      </span>
      <button
        className="consonant-left-filter--footer-btn"
        onClick={[Function]}
        type="button"
      >
        Done
      </button>
    </div>
  </div>
</div>
`;
